flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/FlinkCalciteCatalogReader.java

```diff
@@ -62,17 +83,197 @@ public FlinkCalciteCatalogReader(
 		if (originRelOptTable == null) {
 			return null;
 		} else {
-			// Wrap FlinkTable as FlinkRelOptTable to use in query optimization.
-			FlinkTable table = originRelOptTable.unwrap(FlinkTable.class);
+			// Wrap as linkPreparingTableBase to use in query optimization.
+			CatalogSchemaTable table = originRelOptTable.unwrap(CatalogSchemaTable.class);
 			if (table != null) {
-				return FlinkRelOptTable.create(
-					originRelOptTable.getRelOptSchema(),
-					originRelOptTable.getRowType(),
+				return toPreparingTable(originRelOptTable.getRelOptSchema(),
 					originRelOptTable.getQualifiedName(),
+					originRelOptTable.getRowType(),
 					table);
 			} else {
 				return originRelOptTable;
 			}
 		}
 	}
+
+	/**
+	 * Translate this {@link CatalogSchemaTable} into Flink source table.
+	 */
+	public static FlinkPreparingTableBase toPreparingTable(RelOptSchema relOptSchema,
+			List<String> names,
+			RelDataType rowType,
+			CatalogSchemaTable table) {
+		if (table.isTemporary()) {
+			return convertTemporaryTable(relOptSchema,
```

Comment:
if you are already warping lines for parameters, it's better to wrap the first parameter as well.